Tests/test_Phylo.py

```diff
@@ -91,26 +96,26 @@ def test_newick_read_scinot(self):
 
     def test_phylo_read_extra(self):
         """Additional tests to check correct parsing."""
-        tree = Phylo.read(StringIO("(A:1, B:-2, (C:3, D:4):-2)"), "newick")
+        tree = Phylo.read(StringIO("(A:1,B:-2,(C:3,D:4):-2)"), "newick")
         self.assertEqual(tree.distance("A"), 1)
         self.assertEqual(tree.distance("B"), -2)
         self.assertEqual(tree.distance("C"), 1)
         self.assertEqual(tree.distance("D"), 2)
 
-        tree = Phylo.read(StringIO("((A:1, B:-2):-5, (C:3, D:4):-2)"), "newick")
+        tree = Phylo.read(StringIO("((A:1,B:-2):-5,(C:3,D:4):-2)"), "newick")
         self.assertEqual(tree.distance("A"), -4)
         self.assertEqual(tree.distance("B"), -7)
         self.assertEqual(tree.distance("C"), 1)
         self.assertEqual(tree.distance("D"), 2)
 
-        tree = Phylo.read(StringIO("((:1, B:-2):-5, (C:3, D:4):-2)"), "newick")
+        tree = Phylo.read(StringIO("((:1,B:-2):-5,(C:3,D:4):-2)"), "newick")
         distances = {-4.0: 1, -7.0: 1, 1: 1, 2: 1}
         for x in tree.get_terminals():
             entry = int(tree.distance(x))
             distances[entry] -= distances[entry]
             self.assertEqual(distances[entry], 0)
 
-        tree = Phylo.read(StringIO("((:\n1\n,\n B:-2):-5, (C:3, D:4):-2);"), "newick")
+        tree = Phylo.read(StringIO("((:\n1\n,\nB:-2):-5,(C:3,D:4):-2);"), "newick")
```

Comment:
You're removing the space following the new-line in this and the above similar examples.

Do you think the space was invalid, and ought not to have been accepted?

Reply:
I am no Newick's tree notation expert so please bear with me :p.

As I far as I could understand, unquoted label names cannot contain any UNSAFE_CHARS, where the set of UNSAFE_CHARS is: `[' ', '\n', '\r', '\t', '\f', '\v']`. So, I assumed that any unquoted label name containing any of those chars should be flagged as an invalid name. That is why I decided to remove the whitespace in any unquoted label name, even if it is a leading whitespace. Moreover, all examples on Newick's trees that I have come across avoid using whitespaces at all costs. So, I further assumed that those trees should contain no unnecessary whitespaces.

If you don't agree with me, I can revert the changes I made to the tests. However, in such case I should update the logic that finds invalid chars in an unquoted label name string as well, since it would flag those examples as invalid names (the resulting name would be `•B`).

Reply:
You could be right - perhaps the parser was wrongly tolerating leading whitespace on the continuation lines. If so, an explicit test of that would be best.

@etal are you able to review this please?